websites/C/Coub/presence.ts

```diff
@@ -1,247 +1,370 @@
+          !!ref.location.pathname.match(
+            /^\/(hot|rising|fresh|feed|rising|stories|random|bookmarks|likes|weekly|best|stories|royal\.coubs)/gi
+          ),
+        exec: (
+          context,
+          data,
+          { showWatch, showBridge }: ExecutionArguments
+        ) => {
+          if (!context) return null;
+          const activeMedia = document.querySelector<HTMLElement>(
+            ".coub[coub-block].active"
+          );
+          if (!activeMedia) return null;
+          const title = activeMedia
+              .querySelector(".description__title")
+              .textContent?.trim(),
+            activeTab =
+              document.querySelector<HTMLElement>(
+                ".page__content .page-menu > .page-menu__inner > .page-menu__item.-active"
+              ) ||
+              document.querySelector<HTMLElement>(
+                ".best2020__container .page-menu > .page-menu__inner > .page-menu__item.-active"
+              ) ||
+              document.querySelector<HTMLElement>(
+                ".page__content .story__header"
+              ),
+            isWeekly = !!document.querySelector(
+              ".page__content .page-menu.weekly__menu"
+            ),
+            isBestOfTheYear = location.pathname.startsWith("/best"),
+            isCoubPicks = location.pathname.startsWith("/royal.coubs"),
+            isLiked = !!activeMedia.querySelector(
+              ".coub__like-button[widget-like-button].-on"
+            ),
+            activeTabTitle =
+              activeTab?.textContent?.trim() ||
+              activeTab?.dataset?.title ||
+              "Feed";
+          if (!title) return null;
+          const pageType = document
+            .querySelector(".page__content")
+            ?.getAttributeNames()
+            ?.map((x) => x.match(/^pages-(\w+)-page/i))
+            .filter((x) => !!x && x.length > 1 && x[1])
+            .map((x) => capitalizeFirstLetter(x[1]));
+          data.state =
+            `Browsing ${
+              isCoubPicks && activeTabTitle.match(/^(\w+)/gi)
+                ? activeTabTitle.match(/^(\w+)/gi)[0]
+                : activeTabTitle
+            }` +
+            (pageType?.length > 0 &&
+            activeTabTitle.toLowerCase() !== pageType[0].toLowerCase()
+              ? ` in ${pageType[0]}`
+              : isWeekly
+              ? " in Weekly"
+              : isBestOfTheYear
+              ? " in Best of the Year"
+              : isCoubPicks
+              ? " in Coub Picks"
+              : "");
+          data.details = `${title}${isLiked ? " (❤)" : ""}`;
+          const sourceLink = activeMedia.querySelector<HTMLAnchorElement>(
+            ".description__stamp a.description__stamp__source"
+          );
+          data.buttons = [];
+          if (showWatch)
+            data.buttons.push(
+              ...[
+                ...getSourceLink(sourceLink?.href),
+                {
+                  label: "Watch on Coub",
```

Comment:
Correct me if I'm wrong, but I think for the "Watch on" part of this string (& applies to others) there exists strings you could use since you're using localized strings why not use these. Unless you're never fetching your localized strings from PreMiD and somewhere else, idk.

Reply:
Not worth creating extra localization (unless I'm not needed to add it to translate.premid.app).
Using some strings as fallback like idle/browsing.

Reply:
the "View Page" or the "Watch Video" strings?